File: frontend/src/api/storage.ts
import { apiClient } from "./client.js";

export interface StorageBreakdownItem {
  count: number;
  size_bytes: number;
}

export interface StorageStats {
  total_size_bytes: number;
  total_size_human: string;
  asset_count: number;
  deleted_count: number;
  deleted_size_bytes: number;
  breakdown: {
    images: StorageBreakdownItem;
    meshes: StorageBreakdownItem;
    rigs: StorageBreakdownItem;
    animations: StorageBreakdownItem;
    exports: StorageBreakdownItem;
  };
}

export async function getStorageStats(): Promise<StorageStats> {
  const { data } = await apiClient.get<StorageStats>("/storage/stats");
  return data;
}

export interface PurgeDeletedResponse {
  deleted_count: number;
  freed_bytes: number;
}

export async function purgeDeleted(): Promise<PurgeDeletedResponse> {
  const { data } = await apiClient.post<PurgeDeletedResponse>(
    "/storage/purge-deleted"
  );
  return data;
}

File: api/app/routers/storage.py
"""Storage-API: Statistik und Papierkorb-Purge."""

from fastapi import APIRouter

from app.services.storage_service import compute_storage_stats, purge_deleted

router = APIRouter(prefix="/storage", tags=["storage"])


@router.get("/stats")
async def get_storage_stats():
    """Speicherverbrauch und Breakdown nach Typ (images, meshes, rigs, animations, exports)."""
    return compute_storage_stats()


@router.post("/purge-deleted")
async def purge_deleted_assets():
    """Löscht alle Assets mit deleted_at permanent. Gibt Anzahl und freigegebene Bytes zurück."""
    count, freed = purge_deleted()
    return {"deleted_count": count, "freed_bytes": freed}
